src/main/java/wolox/training/services/BookService.java

```diff
@@ -26,11 +30,16 @@
     @Autowired
     private BookMapper bookMapper;
 
-    public Iterable<Book> findAll(String isbn) {
-        if (StringUtils.isNotEmpty(isbn)) {
-            return filterBookByIsbn(isbn);
-        }
-        return bookRepository.findAll();
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public Iterable<Book> findAll(Map<String,String> params) {
```

Comment:
The idea of the card was to implement a query that ignores null values. For example:
```
SELECT * FROM books
WHERE (:title IS NULL OR title = :title) AND ...
```
But I also like this solution.